src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -249,12 +249,12 @@ private void GenerateCheckBox(ModelExplorer modelExplorer, TagHelperOutput outpu
                 output.Attributes.Clear();
                 output.TagName = null;
 
-                output.Content.Append(tagBuilder.ToString(TagRenderMode.SelfClosing));
+                output.Content.Append(tagBuilder.ToHtmlContent(TagRenderMode.SelfClosing));
```

Comment:
Yeah even though it's not the design I like, it does seem better for perf to change the design to have this implement the interface instead of allocating new objects.